File: scratchpad/failed-jobs/2026-06-02__17-56-57/generate_user_jwt_node__hsCu3v6/artifacts/user/myproject/generate-user-jwt.js
const fs = require("fs");
const jwt = require("jsonwebtoken");

const {
  MAGICBELL_EMAIL,
  MAGICBELL_PROJECT_TOKEN,
  MAGICBELL_API_KEY,
  MAGICBELL_SECRET_KEY,
  ZEALT_RUN_ID,
} = process.env;

if (!MAGICBELL_EMAIL) {
  throw new Error("Missing MAGICBELL_EMAIL environment variable.");
}
if (!MAGICBELL_PROJECT_TOKEN) {
  throw new Error("Missing MAGICBELL_PROJECT_TOKEN environment variable.");
}
if (!MAGICBELL_API_KEY) {
  throw new Error("Missing MAGICBELL_API_KEY environment variable.");
}
if (!MAGICBELL_SECRET_KEY) {
  throw new Error("Missing MAGICBELL_SECRET_KEY environment variable.");
}
if (!ZEALT_RUN_ID) {
  throw new Error("Missing ZEALT_RUN_ID environment variable.");
}

const [emailLocal, emailDomain] = MAGICBELL_EMAIL.split("@");
if (!emailLocal || !emailDomain) {
  throw new Error("MAGICBELL_EMAIL must be a valid email address.");
}

const userEmail = `${emailLocal}+jwt-node-${ZEALT_RUN_ID}@${emailDomain}`;
const userExternalId = `user-jwt-node-${ZEALT_RUN_ID}`;

const upsertUser = async () => {
  const response = await fetch("https://api.magicbell.com/v2/users", {
    method: "PUT",
    headers: {
      Authorization: `Bearer ${MAGICBELL_PROJECT_TOKEN}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      external_id: userExternalId,
      email: userEmail,
    }),
  });

  if (!response.ok) {
    const errorText = await response.text();
    throw new Error(`MagicBell user upsert failed (${response.status}): ${errorText}`);
  }

  return response.json();
};

const generateUserJwt = () =>
  jwt.sign(
    {
      user_email: userEmail,
      user_external_id: userExternalId,
      api_key: MAGICBELL_API_KEY,
    },
    MAGICBELL_SECRET_KEY,
    {
      algorithm: "HS256",
      expiresIn: "1y",
    }
  );

const writeOutput = (token) => {
  const outputLine = `User JWT: ${token}`;
  fs.writeFileSync("/home/user/myproject/output.log", `${outputLine}\n`, "utf8");
};

const main = async () => {
  await upsertUser();
  const token = generateUserJwt();
  writeOutput(token);

  console.log(`Generated User JWT for ${userEmail}`);
};

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: scratchpad/failed-jobs/2026-06-02__17-56-57/generate_user_jwt_python__FUpEwEm/artifacts/user/myproject/generate_user_jwt.py
import json
import os
import time

import jwt
import requests


def get_required_env(name: str) -> str:
    value = os.getenv(name)
    if not value:
        raise RuntimeError(f"Missing required environment variable: {name}")
    return value


def build_user_email(base_email: str, run_id: str) -> str:
    local, domain = base_email.split("@", 1)
    return f"{local}+jwt-python-{run_id}@{domain}"


def build_external_id(run_id: str) -> str:
    return f"user-jwt-python-{run_id}"


def upsert_user(project_token: str, api_key: str, external_id: str, email: str) -> None:
    url = "https://api.magicbell.com/v2/users"
    headers = {
        "Authorization": f"Bearer {project_token}",
        "X-MAGICBELL-API-KEY": api_key,
        "Content-Type": "application/json",
        "Accept": "application/json",
    }
    payload = {
        "external_id": external_id,
        "email": email,
        "first_name": "JWT",
        "last_name": "Python",
    }
    response = requests.put(url, headers=headers, data=json.dumps(payload), timeout=30)
    response.raise_for_status()


def build_user_jwt_payload(email: str, external_id: str, api_key: str) -> dict:
    exp = int(time.time()) + 365 * 24 * 60 * 60
    return {
        "user_email": email,
        "user_external_id": external_id,
        "api_key": api_key,
        "exp": exp,
    }


def main() -> None:
    run_id = get_required_env("ZEALT_RUN_ID")
    base_email = get_required_env("MAGICBELL_EMAIL")
    project_token = get_required_env("MAGICBELL_PROJECT_TOKEN")
    api_key = get_required_env("MAGICBELL_API_KEY")
    secret_key = get_required_env("MAGICBELL_SECRET_KEY")

    user_email = build_user_email(base_email, run_id)
    external_id = build_external_id(run_id)

    upsert_user(project_token, api_key, external_id, user_email)

    payload = build_user_jwt_payload(user_email, external_id, api_key)
    token = jwt.encode(payload, secret_key, algorithm="HS256")

    output_path = "/home/user/myproject/output.log"
    with open(output_path, "w", encoding="utf-8") as handle:
        handle.write(f"User JWT: {token}\n")


if __name__ == "__main__":
    main()
